core/ordersync/ordersync.go

```diff
@@ -449,68 +443,88 @@ func (s *Service) getOrdersFromPeer(ctx context.Context, providerID peer.ID) err
 		_ = stream.Close()
 	}()
 
-	var nextReq *Request
-	var selectedSubprotocol Subprotocol
+	// First request
+	nextReq := &rawRequest{
+		Type:         TypeRequest,
+		Subprotocols: s.SupportedSubprotocols(),
+		Metadata:     nil,
+	}
+	nextReq, err = s.makeOrderSyncRequest(ctx, nextReq, stream, providerID)
+	if err != nil {
+		return err
+	}
+	if nextReq == nil {
```

Comment:
This doesn't look right to me. Just because`nextReq` is nil does not mean we didn't receive any orders from the provider. A successful ordersync will always result in the last `nextReq` being nil.